src/sidebar/groups.js

```diff
@@ -14,17 +14,59 @@
 var STORAGE_KEY = 'hypothesis.groups.focus';
 
 var events = require('./events');
+var { awaitStateChange } = require('./util/state-util');
 
 // @ngInject
-function groups(localStorage, serviceUrl, session, $rootScope, store) {
+function groups(annotationUI, localStorage, serviceUrl, session, $rootScope, store) {
   // The currently focused group. This is the group that's shown as selected in
   // the groups dropdown, the annotations displayed are filtered to only ones
   // that belong to this group, and any new annotations that the user creates
   // will be created in this group.
-  var focusedGroup;
+  var focusedGroupId;
+  var groups = [];
+  var documentUri;
+
+  function getDocumentUriForGroupSearch() {
+    function mainUri() {
+      var uris = annotationUI.searchUris();
+      if (uris.length === 0) {
+        return null;
+      }
+
+      // We get the first HTTP URL here on the assumption that group scopes must
+      // be domains (+paths)? and therefore we need to look up groups based on
+      // HTTP URLs (so eg. we cannot use a "file:" URL or PDF fingerprint).
+      return uris.find(uri => uri.startsWith('http'));
+    }
+    return awaitStateChange(annotationUI, mainUri);
+  }
+
+  /**
+   * Fetch the list of applicable groups from the API.
+   *
+   * The list of applicable groups depends on the current userid and the URI of
+   * the attached frames.
+   */
+  function load() {
+    return getDocumentUriForGroupSearch().then(uri => {
+      return store.groups.list({ document_uri: uri });
+    }).then(gs => {
+      var focGroup = focused();
+      if (focGroup) {
+        var focusedGroupInFetchedList = gs.find(g => g.id === focGroup.id);
+        if (!focusedGroupInFetchedList) {
+          $rootScope.$apply(() => {
+            focus(gs[0].id);
+          });
+        }
+      }
+      groups = gs;
```

Comment:
As you found in your local testing, you need to wrap the `groups` assignment in `$rootScope.$apply`. Since this variable is modified in a Promise callback, and therefore outside of whatever context triggered the `load` call, Angular won't know that it should do a digest cycle at this point.